File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/phone-number/phone_number.py
class PhoneNumber:
    def __init__(self, number):
        # Process characters and validate input format
        digits = []
        has_letters = False
        
        for c in number:
            if c.isdigit():
                digits.append(c)
            elif c.isalpha():
                has_letters = True
        
        if has_letters:
            raise ValueError("letters not permitted")
        
        digits_str = ''.join(digits)
        num_digits = len(digits_str)
        
        # Validate digit count
        if num_digits < 10:
            raise ValueError("must not be fewer than 10 digits")
        if num_digits > 11:
            raise ValueError("must not be greater than 11 digits")
        
        # Validate country code
        if num_digits == 11:
            if digits_str[0] != '1':
                raise ValueError("11 digits must start with 1")
            digits_str = digits_str[1:]
        
        # Validate area code (NXX)
        area_code = digits_str[:3]
        if area_code[0] in ('0', '1'):
            raise ValueError(f"area code cannot start with {'zero' if area_code[0] == '0' else 'one'}")
        
        # Validate exchange code (NXX)
        exchange_code = digits_str[3:6]
        if exchange_code[0] in ('0', '1'):
            raise ValueError(f"exchange code cannot start with {'zero' if exchange_code[0] == '0' else 'one'}")
        
        self.number = digits_str
        self.area_code = digits_str[:3]
        self.exchange_code = digits_str[3:6]
        self.subscriber_number = digits_str[6:]
    def pretty(self):
        return f"({self.area_code})-{self.exchange_code}-{self.subscriber_number}"

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/cpp/exercises/practice/phone-number/phone_number.cpp
#include "phone_number.h"
#include <algorithm>
#include <cctype>

using namespace std;

namespace phone_number {

phone_number::phone_number(const string& raw) {
    string cleaned;
    
    // Remove all non-digit characters
    copy_if(raw.begin(), raw.end(), back_inserter(cleaned), 
        [](char c) { return isdigit(c); });

    // Validate number length and country code
    if (cleaned.length() == 11) {
        if (cleaned[0] != '1') {
            throw phone_number_error("Invalid number length");
        }
        cleaned = cleaned.substr(1);
    }
    
    if (cleaned.length() != 10) {
        throw phone_number_error("Invalid number length");
    }
    
    validate(cleaned);
    
    number_ = cleaned;
    area_code_ = cleaned.substr(0, 3);
    exchange_ = cleaned.substr(3, 3);
    subscriber_ = cleaned.substr(6);
}

void phone_number::validate(const string& number) {
    // Validate area code (NXX)
    if (number[0] < '2' || number[0] > '9') {
        throw phone_number_error("Area code must start with 2-9");
    }
    
    // Validate exchange code (NXX)
    if (number[3] < '2' || number[3] > '9') {
        throw phone_number_error("Exchange code must start with 2-9");
    }
}

}  // namespace phone_number
